dapr_agents/workflow/mixins/state.py

```diff
@@ -0,0 +1,249 @@
+            self.save_state()
+        except Exception as e:
+            raise RuntimeError(f"Error initializing workflow state: {e}") from e
+
+    def validate_state(self, state_data: dict) -> dict:
+        """
+        Validate the workflow state against ``state_format`` if provided.
+
+        Args:
+            state_data: The raw state data to validate.
+
+        Returns:
+            dict: The validated and structured state.
+
+        Raises:
+            ValidationError: If the state data does not conform to the expected schema.
+        """
+        try:
+            if not self.state_format:
+                logger.warning(
+                    "No schema (state_format) provided; returning state as-is."
+                )
+                return state_data
+
+            logger.info("Validating workflow state against schema.")
+            validated_state: BaseModel = self.state_format(**state_data)
+            return validated_state.model_dump()
+        except ValidationError as e:
+            raise ValidationError(f"Invalid workflow state: {e.errors()}") from e
+
+    def load_state(self) -> dict:
+        """
+        Load the workflow state from the configured Dapr state store.
+
+        Returns:
+            dict: The loaded and optionally validated state.
+
+        Raises:
+            RuntimeError: If the state store is not properly configured.
+            TypeError: If the retrieved state is not a dictionary.
+            ValidationError: If state schema validation fails.
+        """
+        try:
+            if (
+                not self._state_store_client
+                or not self.state_store_name
+                or not self.state_key
+            ):
+                logger.error("State store is not configured. Cannot load state.")
+                raise RuntimeError(
+                    "State store is not configured. Please provide 'state_store_name' and 'state_key'."
+                )
+
+            if self.state:
+                logger.info(
+                    "Using existing in-memory state. Skipping load from storage."
+                )
+                return self.state
+
+            has_state, state_data = self._state_store_client.try_get_state(
+                self.state_key
+            )
+            if has_state and state_data:
+                logger.info(
+                    f"Existing state found for key '{self.state_key}'. Validating it."
+                )
+                if not isinstance(state_data, dict):
+                    raise TypeError(
+                        f"Invalid state type retrieved: {type(state_data)}. Expected dict."
+                    )
+                return (
+                    self.validate_state(state_data) if self.state_format else state_data
+                )
+
+            logger.info(
+                f"No existing state found for key '{self.state_key}'. Initializing empty state."
```

Comment:
should these be debug level?

Reply:
Yeah those could be debug. It is hard to figure out which is info and debug tbh. 😅

Reply:
I will focus on logging in a separate PR. I would like to brainstorm a little bit more the strategy for that based on your experience building SDKs ;) Thank you!